Automattic/studio
# Studio: Improve offline mode for Sync tab

## Related issues

Closes https://github.com/Automattic/dotcom-forge/issues/9808

## Proposed Changes

This PR handles offline mode for the `Sync` tab.

**On the modal: if the connection goes out when the user has the modal open**:

<img width="656" alt="Screenshot 2024-11-15 at 3 21 37 PM" src="https://github.com/user-attachments/assets/97960cfa-a1a7-4503-b445-4e6a6846ef25">

**On the Sync tab itself**:

<img width="476" alt="Screenshot 2024-11-15 at 3 21 53 PM" src="https://github.com/user-attachments/assets/59f6d532-c81d-4b88-b412-3d8d98e0c83e">

<img width="301" alt="Screenshot 2024-11-15 at 3 22 00 PM" src="https://github.com/user-attachments/assets/37c1c154-4867-49ad-80a5-9d5dbe406764">


## Testing Instructions



* Pull the changes from this branch
* Start Studio app with `STUDIO_SITE_SYNC=true npm start`
* Navigate to the `Sync` tab that has some sites connected 
* Disable your internet connection (in my case, it was enough to disconnect from WiFi)
* Confirm that the buttons to `Create a new site` and `Connect site` are disabled
* Confirm that when you hover over `Pull` or `Push` buttons, they are disabled and you can see a tooltip with information
* Enable your internet connection
* Click on `Connect site` to open a modal
* Disable your internet connection
* Observe that you see the notice that lays over the content and does not allow select or connect sites
* Repeat the same steps on a Studio site that has no WP.com sites connected to it and confirm the buttons to connect a site or create a new site are disabled

## Pre-merge Checklist
